### PR TITLE
Add WellTestState object to keep track of automatically closed wells

### PR DESCRIPTION
**Pull request by joakim-hove**
This object should be instantiated and managed from the simulator; dangling here.

@totto82 : I hope this seems usable; please suggest changes.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/parser/eclipse/EclipseState/Schedule/WellTestState.hpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        double last_test;
+        int num_attempt;
+    };
+
+    /*
+      The simulator has decided to close a particular well; we then add it here
+      as a closed well with a particualar reason.
+    */
+    void addClosedWell(const std::string& well_name, WellTestConfig::Reason reason, double sim_time);
+
```
Comment (totto82) [inline opm/parser/eclipse/EclipseState/Schedule/WellTestState.hpp]: Why provide the sim_time here? Isn't it the timing between the testing that is used.

**Comment by totto82** [comment]
Thanks. Maybe I will change it a bit while implementing the simulator part. It is not 100% clear how to best implement the WTEST feature from the simulator side.

File: src/opm/parser/eclipse/EclipseState/Schedule/WellTestState.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    std::vector<std::pair<std::string, WellTestConfig::Reason>> WellTestState::update(const WellTestConfig& config, double sim_time) {
+        std::vector<std::pair<std::string, WellTestConfig::Reason>> output;
+        for (auto& closed_well : this->wells) {
+            if (config.has(closed_well.name, closed_well.reason)) {
+                const auto& well_config = config.get(closed_well.name, closed_well.reason);
+                double elapsed = sim_time - closed_well.last_test;
+
+                if (elapsed >= well_config.test_interval)
+                    if (closed_well.num_attempt < well_config.num_test) {
```
Comment (totto82) [inline src/opm/parser/eclipse/EclipseState/Schedule/WellTestState.cpp]: if well_config.num_test = 0 the well should be tested unlimited number of times.

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/WellTestState.cpp]: Ahhh  ok - will fix.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
It will be very helpful when adding closed wells or completions to WellTestState, some message about the action can be given.  Currently, everything happens silently.

**Comment by GitPaean** [comment]
> Currently, everything happens silently.

I might be wrong.  Sorry about the noise.

**Pull request merged**